### DISCUSSION TITLE
Search doesn't search in image description

### DISCUSSION THREAD

**Discussion by akoscomp** [immich-app/immich#7823] [Feature Request]
### The bug

When I add a description to an image and I try to search for added description, I don't got the image with description.

### The OS that Immich Server is running on

docker

### Version of Immich Server

v1.98.1

### Version of Immich Mobile App

v1.98.1

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
standard
```

### Your .env content

```Shell
standard
```

### Reproduction steps

```bash
1. Add a description
2. Search for one word from description
3. I don't got the image with description
...
```

**Comment by zenaufa** [comment]
This issue also happened to me. I've already tried to change the Smart Search CLIP model to "immich-app/nllb-clip-large-siglip__v1" and to the smart search job for all the images, but when i tried to search by the word that i put in an image descrpition, those image doesn't come up.

**Comment by rodnheli** [comment]
Same here - searching for a description doesnt't work (would also like to find an album when i search the album name)
docker - [v1.103.1]

**Comment by SwallowYourDreams** [comment]
Issue is still present in 1.113.0 on docker. Needs fixing.

**Reply by bo0tzz** [reply]
This is a feature request, not a bug, so "Needs fixing." doesn't apply and comes off as entitled.

**Reply by SwallowYourDreams** [reply]
The issue is phrased like a bug report, hence my wording. Tomato, tomato. Either way, no offense intended.

**Comment by Acidiphilium** [comment]
Agree, not being able to find description or in general the lack of search options is (at least for me), one last missing feature which holds me to switch definitely from Google Photos. Would love to see some search improvements.

**Comment by johnicepick** [comment]
I'm baffled that I cannot search for photos that contain description or keywords. 
I'm looking for a platform where I can place like 1000-2000 photos of our gliding club. All photos are tagged with the names of the people that are on the photo, face recognition won't help here, since some of the people are sitting inside planes. Or searching for that one plane modell in front of a blue sky. I've already written that info.

Such a sophisticated system, with face recognition, but no ability to serach for EXIF/ IPTC fields. Baffling, it makes me speechless.
I'm working at a press agency, and I'm writing IPTC descriptions to hundreds of photos everyday, so that other newspapers, magazines can find them. Having searchable photos is the -core- function of a photodatabase - I truly hope that immich will have that one day.

**Reply by ok1ok1** [reply]
The limitations with search are strange. Also odd that this feature is not on the roadmap. I realize not every feature makes the roadmap but this seems like a big one.

**Reply by alextran1502** [reply]
We are working on database refactoring, this will probably have to wait until that is done first.

**Reply by johnicepick** [reply]
@bo0tzz & @alextran1502 yet smartsearch or the search for filenames is working. Please put the search for description for the for the time being in a provisional way in it. And do the re-work later.
Example left: IPTC for pressphotos for databases like Getty Images, on the right: minimalistic description for my glider club to find images of themselfs back how they piloting their planes. All that information is completly ignored. Bummer.
![Screenshot 2024-11-25 152318 2](https://github.com/user-attachments/assets/3a38656d-f567-453e-a255-4182192f9ae1)


Also smart search won't help it, there people who are strictly against any kind of AI Recognition, all of that is disabled. And even if it works, I have several hundreds of photos who look like this. This only works with description search!
![Screenshot 2024-11-25 152720](https://github.com/user-attachments/assets/c36dcdaf-0407-4764-b580-e04909c9fe96)

@ok1ok1 searching your photos is such a basic function that I wouldn't even think about putting it on a roadmap. Especially if AI face detection is already implemented. Its like building an airplane, but forgetting the chairs to sit on. Immich is a fantastic system, and I will pay for the server license, because you really did an awesome job here - but without a search for description function...? 
I have to tell my flying mates with: yes I host the images, several thousands, but you cannot search for them. Yes it has a search bar, but its not functional. When will it be? I dunno....

**Reply by alextran1502** [reply]
@johnicepick we are pausing some
of the database work because we are migrating the codebase to a different ORM library. When that is done, we can add this in

You can track in #12857 for the database work migration

**Reply by johnicepick** [reply]
@alextran1502 thanks for your response. I have a feeling that this is a lengthy proces, yet I hope Kysely will be soon implemented. (And with soon I don't mean like in 6 months or so ...). Because I'm not the only one who "suffers" from not having searchable descriptions

**Reply by alextran1502** [reply]
Kysley is our high priority task at the moment

**Comment by akostadinov** [comment]
Yes, it came as a surprise that simple search in description is not supported. Actually this is an important use case as several things like "washing machine warranty", "my plate number", "piping in the kitchen", etc. are extremely useful... to find images with important information.

It doesn't seem humanly possible to me to remember the time I took the picture. Fortunately I can still use google photos for this but it is also one of the reasons I don't want to use google photos.

I think it makes sense to prioritize this feature due to the high interest as it will increase adoption and is probably not as hard as reorganizing the database. While users can be annoying, but I hope adoption is the goal ¯\_(ツ)_/¯

**Reply by bo0tzz** [reply]
Not to diminish your point - but try putting those queries into the smart search. It should at the very least get pretty close.

**Reply by johnicepick** [reply]
@bo0tzz yes and no. I don't think that smart search would recognize the difference of piping in the kitchen or in the bathroom. Pipes = pipes, and if he lays lots of pipes (pun not intended) than smart search would spit out all dozens, hundreds of pipes.

**Comment by Knuppel1983** [comment]
Same problem here. Disabled machine learning (don’t need it, plus too much strain on my nas). Added some tags and description. Search for tags. Nothing comes up. What? This is basic stuff?

I have a large amount of photos cut up in subfolders. My plan is to tag the files with the subfoldername and then import them into Immich. Right now i don’t know whether to go with relatively mature and stable Synology Photos or take the plunge with Immich. The latter showing more potential imho, being multi-platform.

I know the world is all about AI and machine learning today, but please don’t forget there are ppl deliberately trying to move or stay away from Google Photos, because they don’t want AI. They just want a photo library that they can control. One that absorbs previous tagging and doesn’t fall into pieces when they decide to migrate.

Maybe put up a poll? Move attention to developing and improving the basics, then go back to machine learning? I think maturing the product first would be the wisest move, giving users confidence to trust their precious photos with Immich.

(kudos for the work though, i know the amount of time that goes into this)

**Comment by pavlinbl4** [comment]
This gallery is useless to me without a basic search by image description and keywords.

**Comment by D4ndellion** [comment]
In the roadmap it says 

```
# 3/29/2023 - Search by metadata [[v1.52.0]](https://github.com/immich-app/immich/releases/tag/v1.52.0)

Search images by filename, description, tagged people, make, model, and other metadata.
```

so I was initially very confused why I couldn't get it to work. Would be nice to have this!

**Reply by alextran1502** [reply]
Yeah we had this previously when we were using [Typesense](https://typesense.org/) for search. As we simplify the deployment stack, it is temporarily removed with that. We will put it back after database refactoring work that we are working on

**Reply by Knuppel1983** [reply]
> Yeah we had this previously when we were using [Typesense](https://typesense.org/) for search. As we simplify the deployment stack, it is temporarily removed with that. We will put it back after database refactoring work that we are working on

Can you give us a rough estimate when this feature will be back? Would help in deciding which bag to put my eggs in.

**Reply by bo0tzz** [reply]
There is no estimate

**Comment by zboina** [comment]
I'm waiting for this functionality! When it appears, I will use your application to manage our photo database!
Is there a chance that this functionality will appear in the near future?

**Reply by bo0tzz** [reply]
Did you read the comment above yours?

**Reply by zboina** [reply]
Yes, but I wanted to emphasize that this is an interesting and necessary functionality :-)
Maybe you will implement it faster this way :-)
A big thank you to the whole team for a great application!

**Reply by bo0tzz** [reply]
>  Maybe you will implement it faster this way :-)

Nope, that's not how it works.

**Comment by alextran1502** [comment]
Implemented in `v1.126.0`

**Reply by zboina** [reply]
Great! Thank you very much! If only we could choose a default search type... Now it's context.

**Reply by akostadinov** [reply]
Yeah, thank you. This feature was a real bummer. Will try it out over the weekend.

**Reply by xela92** [reply]
This feature is super awesome, just tried it and works perfectly 🔥

**Comment by pjfitzgerald** [comment]
Appreciate the work on this! It seems as of now when searching for content in a description (or any search for that matter) the default behaviour is to not search archived photos. Wondering if a better solution would be to have the display options 'Archive' option be changed to 'include archive' (and be on by default) which would then return all photos matching the search both archived and unarchived? At least this is the approach I would prefer - at it stands it seems like we have to search twice to cover archived/unarchived results.

**Reply by alextran1502** [reply]
<img width="1015" height="969" alt="image" src="https://github.com/user-attachments/assets/038b9c26-16a2-4a5d-b809-2b2799856ccb" />

You can do this in one go in the search filter here

**Reply by pjfitzgerald** [reply]
Hey Alex, thanks for the response. I noticed this, but by ticking archive it seems to *only* search archived photos - I'm wondering if searching both the timeline and archive would be better behaviour? At least it would in my case.
